node/src/types/deploy.rs

```diff
@@ -332,6 +369,40 @@ impl Deploy {
         &self.session
     }
 
+    /// Returns true iff:
+    ///   * the deploy hash is correct (should be the hash of the header), and
+    ///   * the body hash is correct (should be the hash of the body), and
+    ///   * the approvals are all valid signatures of the deploy hash
+    ///
+    /// Note: this is a relatively expensive operation, requiring re-serialization of the deploy,
+    ///       hashing, and signature checking, so should be called as infrequently as possible.
```

Comment:
If this method is CPU expensive, maybe instead of making it public on an instance of `Deploy` (which makes it possible to be called by everyone anywhere anytime) we should add it to `DeployAcceptor` (`fn is_valid(&deploy) -> Result<Error, bool>`, note that I added `Error` there to differentiate between different type of validity check failures).

Reply:
Yes - it's a good point.

I believe @EdHastingsCasperLabs is going to do a PR to my branch though where the validity check result is memoized within the Deploy, so I'll hang fire for now until that PR comes in.